MLS.Agent/DotnetKernelJupyterInstaller.cs

```diff
@@ -70,29 +70,16 @@ private static void Installkernel(IEnumerable<DirectoryInfo> dataDirectories, IC
             }
         }
 
-        public static async Task<IEnumerable<DirectoryInfo>> GetJupyterDataPaths(IReadOnlyCollection<string> jupyterPathInfo)
+        public static async Task<IEnumerable<DirectoryInfo>> GetJupyterDataPaths(string[] jupyterPathInfo)
         {
-            var indexOfDataNode = Array.FindIndex(jupyterPathInfo.ToArray(), element => element.TrimLineEndings().CompareTo("data:") == 0);
-            if (indexOfDataNode != -1)
+            var dataHeaderIndex = Array.FindIndex(jupyterPathInfo, element => element.Trim().CompareTo("data:") == 0);
+            if (dataHeaderIndex != -1)
             {
-                var currentPaths = new List<string>();
-                for (var index = indexOfDataNode + 1; index < jupyterPathInfo.Count; index++)
-                {
-                    string element = jupyterPathInfo.ElementAt(index);
-                    if (!string.IsNullOrWhiteSpace(element))
-                    {
-                        if (element.TrimLineEndings().EndsWith(":"))
-                        {
-                            break;
-                        }
-                        else
-                        {
-                            currentPaths.Add(element);
-                        }
-                    }
-                }
+                var nextHeaderIndex = Array.FindIndex(jupyterPathInfo, dataHeaderIndex + 1, element => element.Trim().EndsWith(":"));
+                if (nextHeaderIndex == -1)
+                    nextHeaderIndex = jupyterPathInfo.Count();
 
-                return currentPaths.Select(dir => new DirectoryInfo(dir));
+                return jupyterPathInfo.Skip(dataHeaderIndex+1).Take(nextHeaderIndex - dataHeaderIndex - 1).Select(dir => new DirectoryInfo(dir.Trim()));
```

Comment:
`string.Substring` is much more efficient than using LINQ for string operations.